crates/rnote-ui/src/workspacebrowser/filerow/actions/duplicate.rs

```diff
@@ -12,18 +12,23 @@ use tracing::debug;
 
 /// The regex used to search for duplicated files
 /// ```text
-/// - Look for delimiter
+/// - Look for original filename lazily
 /// |         - Look for `<delim>1`/`<delim>2`/`<delim>123`/...
-/// |         |        - Look for the rest after the `<delim><num>` part
+/// |         |        - Look for the file extension after the `<delim><num>` part
 /// |         |       |       - At the end of the file name
 /// |         |       |       |
-/// |        \d*      |       $
+/// |         (?P<delim{}\d+)$
 /// |                 |
-/// DELIM       (?P<rest>(.*))
+/// ^(.*?)            (\.\w+)
 /// ```
 static DUP_REGEX: Lazy<Regex> = Lazy::new(|| {
-    Regex::new(&(crate::utils::FILE_DUP_SUFFIX_DELIM_REGEX.to_string() + r"\d*(?P<rest>(.*))$"))
-        .unwrap()
+    // Incorporate FILE_DUP_SUFFIX_DELIM_REGEX dynamically into the regex pattern
+    let regex_pattern: String = format!(
+        r"^(.*?)(?P<delim>{}\d*)?(\.\w+)?$",
```

Comment:
Why not keep naming the capture groups? I think it improves readability when using the regex